include/zephyr/bluetooth/gatt.h

```diff
@@ -1007,18 +1007,60 @@ struct bt_gatt_notify_params {
 int bt_gatt_notify_cb(struct bt_conn *conn,
 		      struct bt_gatt_notify_params *params);
 
-/** @brief Notify multiple attribute value change.
- *
- *  This function works in the same way as @ref bt_gatt_notify_cb.
- *
- *  @param conn Connection object.
- *  @param num_params Number of notification parameters.
- *  @param params Array of notification parameters.
- *
- *  @return 0 in case of success or negative value in case of error.
- */
-int bt_gatt_notify_multiple(struct bt_conn *conn, uint16_t num_params,
-			    struct bt_gatt_notify_params *params);
+/** @brief Send multiple notifications in a single PDU.
+ *
+ *  The GATT Server will send a single ATT_MULTIPLE_HANDLE_VALUE_NTF PDU
+ *  containing all the notifications passed to this API.
+ *
+ *  All `params` must have the same `func` and `user_data` (due to
+ *  implementation limitation). But `func(user_data)` will be invoked for each
+ *  parameter.
+ *
+ *  As this API may block to wait for Bluetooth Host resources, it is not
+ *  recommended to call it from a cooperative thread or a Bluetooth callback.
+ *
+ *  The peer's GATT Client must write to this device's Client Supported Features
+ *  attribute and set the bit for Multiple Handle Value Notifications before
+ *  this API can be used.
+ *
+ *  Only use this API to force the use of the ATT_MULTIPLE_HANDLE_VALUE_NTF PDU.
+ *  For standard applications, `bt_gatt_notify_cb` is preferred, as it will use
+ *  this PDU if supported and automatically fallback to ATT_HANDLE_VALUE_NTF
+ *  when not supported by the peer.
+ *
+ *  This API has an additional limitation: it only accepts valid attribute
+ *  references and not UUIDs like `bt_gatt_notify` and `bt_gatt_notify_cb`.
+ *
+ *  @param conn
+ *    Target client.
+ *    Notifying all connected clients by passing `NULL` is not yet supported,
+ *    please use `bt_gatt_notify` instead.
+ *  @param num_params
+ *    Element count of `params` array. Has to be greater than 1.
+ *  @param params
+ *    Array of notification parameters. It is okay to free this after calling
+ *    this function.
+ *
+ *  @retval 0
+ *    Success. The PDU is queued for sending.
+ *  @retval -EINVAL
+ *    - One of the attribute handles is invalid.
+ *    - Only one parameter was passed. This API expects 2 or more.
+ *    - Not all `func` were equal or not all `user_data` were equal.
+ *    - One of the characteristics is not notifiable.
+ *    - An UUID was passed in one of the parameters.
+ *  @retval -ERANGE
+ *    - The notifications cannot all fit in a single ATT_MULTIPLE_HANDLE_VALUE_NTF.
+ *    - They exceed the MTU of all open ATT bearers.
+ *  @retval -EPERM
+ *    The connection has a lower security level than required by one of the
+ *    attributes.
+ *  @retval -EOPNOTSUPP
+ *    The peer hasn't yet communicated that it supports this PDU type.
+ */
+int bt_gatt_notify_multiple(struct bt_conn *conn,
+			    struct bt_gatt_notify_params params[],
+			    uint16_t num_params);
```

Comment:
It's good that it now follows the convention that array size comes after the array, but if you modify the public API like here, we also need to make a stable API change RFC. 

I suggest that since this change of order of argument is not necessary, we revert this change and keep the API as is.

Reply:
fair enough, where is the stable API list ?

Reply:
> fair enough, where is the stable API list ?

I don't think we have a list; I guess any feature that isn't marked with `[EXPERIMENTAl]` is consider stable, right @jhedberg?